A few fixes to make editing DS in the backend work again

cweiske opened the pull request:
When editing TV datastructures in the TYPO3 backend, things get broken or do not work correctly.

Here are some patches that improve datastructures modification.

cweiske committed:
Make datastructure editor use the new "type" tag location

See https://docs.templavoila.support/Changelog/7.6/Breaking-ConfigurationOfTypesInFlexforms.html

cweiske committed:
Copy TV titles to TCEforms labels when editing elements

TYPO3's form engine reads the field labels from TCEforms/label, but
the new templavoila version did not copy the tx_templavoila/title
anymore. The old TV version did that correctly.

cweiske committed:
Correctly pre-select the "section" type when editing a section from an old datastructure

In the previous TV version, sections did not have a tx_templavoila/type value.
We work around that.

cweiske committed:
Work around "integer section" bug in compatibility6

Extension "compatibility6" version 7.6.4 only works with string-typed
section tag values "<section>1</section>".

TV generated "<section type="integer">1</section>" which was
not recognized correctly.

compatibility6 patch:
 https://github.com/FriendsOfTYPO3/compatibility6/pull/14

cweiske committed:
Make container titles available to form engine

Form engine expects container and section titles in a title tag directly
in the element, while templavoila stores it in tx_templavoila/title.
We simply copy that now.

alexanderschnitzler commented:
Thanks a bunch. Please give me some time to review this. I am in a quite time consuming project right now. Will be finished in about 2 weeks.

cweiske committed:
Allow removing items from TCEforms configuration during update

When changing element configuration in the backend editor,
it was not possible to remove items from the form configuration.

To solve that problem we remove the old config completely.

Resolves: https://github.com/alexanderschnitzler/templavoila/issues/11
